src/note_ends.lua

```diff
@@ -63,19 +63,19 @@ function expand_note_ends()
             and entry.Duration + entry:Next().Duration >= note_value
             and
             (    (beat_duration >= position_in_beat + note_value) -- enough space for expanded endpoint
-             or  (not is_compound_meter and not eighth_notes and start_beat % 2 == 0) -- special case quarter note on an odd beat
+                or  (not is_compound_meter and not eighth_notes and start_beat % 2 == 0) -- special case quarter note on an odd beat
             )
-        then
-            local duration_with_rest = entry.Duration + entry:Next().Duration
-            entry.Duration = note_value	-- expand target note
-            if duration_with_rest == note_value then
-                should_delete_next = true -- just delete the following rest
-            elseif duration_with_rest > note_value then -- some duration left over
-                entry:Next().Duration = duration_with_rest - note_value -- make rest smaller
-                should_delete_next = false -- and don't delete it
-            end
+                then
+                local duration_with_rest = entry.Duration + entry:Next().Duration
+                entry.Duration = note_value	-- expand target note
+                if duration_with_rest == note_value then
+                    should_delete_next = true -- just delete the following rest
+                elseif duration_with_rest > note_value then -- some duration left over
+                    entry:Next().Duration = duration_with_rest - note_value -- make rest smaller
+                    should_delete_next = false -- and don't delete it
+                end
         end
```

Comment:
So now you went one tab too far. The final `end` statements should be a procession of tab stops:

```lua
                end
            end
        end
    end
end
```

No gaps and no vertical alignement or overlaps.

Reply:
My last submission PR# acb1acb now looks balanced on GitHub.

Reply:
I'm sorry to be a pain, but this is important. The `then` and `end` are now aligned correctly, but the lines of code *between* them need an indent.